### DISCUSSION TITLE
n00b Q: Server Action Redirects

### DISCUSSION THREAD

**Discussion by arm-learning** [amannn/next-intl#777] [Q&A] [answered]
I am unable to successfully redirect from a server action, is this an appropriate pattern for next-intl? 

I am using redirect that is exported from `navigation` (e.g., `redirect('/')`, and looking into the middleware code, it looks like the rewrite function, specifically this [line](https://github.com/amannn/next-intl/blob/main/packages/next-intl/src/middleware/middleware.tsx#L79C7-L79C62)
```
server.NextResponse.rewrite(urlObj, getResponseInit());
```
OR:
```
return NextResponse.rewrite(urlObj, getResponseInit());
```

results in the URL to be lost:
```
  response: {
  cookies: ResponseCookies {},
  url: '',
  ...
```

super open to any suggestions!

**Comment by amannn** [comment, marked as answer]
I think that's a bug in Next.js: https://github.com/vercel/next.js/issues/58281

**Reply by arm-learning** [reply]
wow, great find! i actually looked through the next.js issues when i ran into this but totally missed that one! thank you for the reply and hope it can be fixed upstream asap!

**Comment by peyronoscar** [comment]
Did you find any solution to this @arm-learning?

**Comment by arm-learning** [comment]
@peyronoscar, sorry i don't remember the full context for this bug, but i am able to redirect now from server actions using next-intl on v4@latest using the redirect provided by next intl `routing` with self hosted nextjs. On apple silicon the dev server sometimes fails to redirect (with next-intl or not), not sure if thats applicable here or what you are seeing, hope you found a resolution!

I will go ahead and close since the issue for me is resolved, feel free to open another discussion if you are still blocked
